lib/datadog/ci/contrib/minitest/hooks.rb

```diff
@@ -19,41 +18,37 @@ def before_setup
             path, = method(name).source_location
             test_suite = Pathname.new(path.to_s).relative_path_from(Pathname.pwd).to_s
 
-            span = CI::Recorder.trace(
-              configuration[:operation_name],
-              {
-                span_options: {
-                  resource: test_name,
-                  service: configuration[:service_name]
-                },
-                framework: Ext::FRAMEWORK,
-                framework_version: CI::Contrib::Minitest::Integration.version.to_s,
-                test_name: test_name,
-                test_suite: test_suite,
-                test_type: Ext::TEST_TYPE
-              }
+            test_span = CI.start_test(
+              test_name,
+              tags: {
+                CI::Ext::Test::TAG_FRAMEWORK => Ext::FRAMEWORK,
+                CI::Ext::Test::TAG_FRAMEWORK_VERSION => CI::Contrib::Minitest::Integration.version.to_s,
+                CI::Ext::Test::TAG_TYPE => Ext::TEST_TYPE,
+                CI::Ext::Test::TAG_SUITE => test_suite
+              },
+              service_name: configuration[:service_name],
+              operation_name: configuration[:operation_name]
             )
 
-            Thread.current[:_datadog_test_span] = span
+            Thread.current[:_datadog_test_span] = test_span
```

Comment:
A bit of an unrelated comment, but this `before_setup` method cannot be nested with itself right? I ask because setting `Thread.current[:_datadog_test_span] = test_span` twice before calling the respective `after_teardown` in the correct order will cause issues.

Reply:
good point, I will check that - should not be the case, I have a bunch of minitest projects for integration testing, did not encounter lost test traces

Reply:
anyway this is something to replace with `Datadog::CI::Context` in the subsequent PR